### PR TITLE
allow to custom region-split-check-diff

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What problem does this PR solve?
Issue Number: close https://github.com/tikv/tikv/issues/9327.

### Check List 

Tests 
- Unit test

Side effects

### Release note 
* No release note.

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -3631,4 +3629,12 @@ mod tests {
             );
         }
     }
+
+    #[test]
+    fn test_validate_tikv_config() {
+        let mut cfg = TiKvConfig::default();
+        cfg.raft_store.region_split_check_diff = ReadableSize::mb(20);
```
Comment (gengliqi) [inline src/config.rs:3636]: How about setting it to `cfg.raft_store.region_split_check_diff + 1`? Thus we can ensure this value is different from the original one.

**Review by gengliqi — COMMENTED** [review]
Rest LGTM

**Review by rleungx — APPROVED** [review]

**Review by gengliqi — APPROVED** [review]
LGTM

**Pull request merged**